specification/devcenter/DevCenter/DevBox/models.tsp

```diff
@@ -682,3 +780,182 @@ model DevBoxRepairOperationResult {
+}
+
+@doc("Parameters for a customization task.")
+@added(APIVersions.v2024_05_01_preview)
+model CustomizationTaskDefinitionParameter {
+  @doc("Description of the parameter.")
+  description?: string;
+
+  @doc("Type of the parameter.")
+  type: CustomizationTaskDefinitionParameterType;
+
+  @doc("Whether or not the parameter is required.")
+  @visibility("read")
+  required?: boolean;
+
+  @doc("Default value for the parameter.")
+  default?: string;
+
+  @doc("Allowed values for the parameter.")
+  allowed?: string[];
+}
+
+@doc("Represents a list of tasks to apply to a Dev Box")
+@added(APIVersions.v2024_05_01_preview)
+model CustomizationTaskList {
+  @doc("Tasks to apply.")
+  tasks?: CustomizationTask[];
+}
+
+@doc("A customization task to run on a Dev Box.")
+@resource("customizationTasks")
+@parentResource(Catalog)
+@added(APIVersions.v2024_05_01_preview)
+model CustomizationTask {
+  @doc("Name of the task.")
+  @key("taskName")
+  @visibility("read", "create")
+  name: string;
+
+  @doc("Parameters for the task.")
+  parameters?: Record<string>;
+
+  @doc("Display name to help differentiate multiple instances of the same task.")
+  displayName?: string;
+
+  @doc("Timeout, in seconds. Overrides any timeout provided on the task definition.")
+  timeoutInSeconds?: int32;
+
+  @doc("What account to run the task as.")
+  runAs?: CustomizationTaskExecutionAccount;
+
+  @doc("ID of the task instance.")
+  @visibility("read")
+  id?: string;
+
+  @doc("The unique URI for retrieving the task logs.")
+  @visibility("read")
+  logUri?: string;
+
+  @doc("Status of the task.")
+  @visibility("read")
+  status?: CustomizationTaskStatus;
+
+  @doc("Start time of the task.")
+  @visibility("read")
+  startTime?: utcDateTime;
+
+  @doc("End time of the task.")
+  @visibility("read")
+  endTime?: utcDateTime;
+}
+
+@doc("The operation result of validating a list of customization tasks.")
+@added(APIVersions.v2024_05_01_preview)
+model CustomizationTaskListValidationOperationResult {
+  ...OperationStatus;
+
+  @doc("Outcome of validation.")
+  result?: CustomizationTaskListValidationResult;
+}
+
+@doc("The result of validating a list of customization tasks.")
+@added(APIVersions.v2024_05_01_preview)
+model CustomizationTaskListValidationResult {
+  @doc("Outcome of validation.")
+  validationResult: CustomizationTaskListValidationStatus;
+
+  @doc("List of validation errors. Absent if no errors.")
+  errors?: CustomizationTaskListValidationError[];
+}
+
+@doc("All of the validation errors for a customization task.")
+@added(APIVersions.v2024_05_01_preview)
+model CustomizationTaskListValidationError {
+  @doc("The customization task that the error is about.")
+  target: CustomizationTask;
+
+  @doc("List of validation errors for the task.")
+  details: CustomizationTaskListValidationErrorDetails[];
+}
+
+@doc("A single error from validating a task.")
+@added(APIVersions.v2024_05_01_preview)
+model CustomizationTaskListValidationErrorDetails {
```

Comment:
Cant we use the Azure.Core error type instead of a custom one?

Reply:
Spoke with designer of this API and confirmed with him that we don't have any reason to not use a standard error object, so I'll make this change.